cmake/sdks.cmake

```diff
@@ -214,10 +213,30 @@ function(add_sdks)
         endif()
 
         add_subdirectory("${SDK_DIR}")
+        message(STATUS "exporting ${SDK_TARGET}")
         LIST(APPEND EXPORTS "${SDK_TARGET}")
         unset(SDK_TARGET)
     endforeach()
 
+    file(GLOB subdirs LIST_DIRECTORIES true "${CMAKE_SOURCE_DIR}/generated/smoke-tests/*")
+    foreach(subdir ${subdirs})
```

Comment:
The code honors the client filter because it looks for the client in sdk_build_list first and only then builds it

 list(FIND SDK_BUILD_LIST ${folder_name} index)

        # Check if the item was found (index >= 0)
        if(${index} GREATER -1)
            message(STATUS "${subdir} is in SDK_BUILD_LIST at index ${index}")

            if(EXISTS "${subdir}/CMakeLists.txt")
                add_subdirectory(${subdir})
            endif()
        else()
            message(STATUS "${subdir} is NOT in SDK_BUILD_LIST")
        endif()